storage/migrations/02_runtimes.up.sql

```diff
@@ -190,7 +190,10 @@ CREATE TABLE chain.evm_token_analysis  -- Moved to analysis.evm_tokens in 06_ana
 (
   runtime runtime NOT NULL,
   token_address oasis_addr NOT NULL,
-  total_supply uint_numeric, -- Dead reckons total_supply before token is downloaded.
+  -- Dead reckons total_supply before token is downloaded. 
+  -- Note that we do not use the custom uint_numeric defined in 01_consensus.up.sql because some blocks 
+  -- may contain negative total_supply updates.
+  total_supply NUMERIC(1000,0), 
```

Comment:
> can be circumvented by typecasting the $3 explicitly
Neat, I didn't think of that. I'm leaning towards the first option so that we keep an accurate tally even for 'misbehaving' contracts instead of overwriting the data we see on-chain. This would require changing the `chain.evm_tokens.total_supply` column as well.

Reply:
sounds good